ScanCommand/src/main/java/io/ballerina/scan/utils/ScanUtils.java

```diff
@@ -120,6 +130,124 @@ private static Target getTargetPath(Project project, String directoryName) {
         }
     }
 
+    /**
+     * Returns the {@link Path} of the html analysis report where generated issues are saved.
+     *
+     * @param issues        generated issues
+     * @param project       Ballerina project
+     * @param directoryName target directory name
+     * @return path of the html analysis report where generated issues are saved
+     */
+    public static Path generateScanReport(List<Issue> issues, Project project, String directoryName) {
+        JsonObject jsonScannedProject = new JsonObject();
+        jsonScannedProject.addProperty("projectName", project.currentPackage().packageName().toString());
+        Map<String, JsonObject> jsonScanReportPathAndFile = new HashMap<>();
+        issues.forEach((issue) -> {
+            IssueImpl issueIml = (IssueImpl) issue;
+            String filePath = issueIml.filePath();
+
+            JsonObject jsonScanReportFile;
+            if (!jsonScanReportPathAndFile.containsKey(filePath)) {
+                jsonScanReportFile = new JsonObject();
+                jsonScanReportFile.addProperty("fileName", issueIml.fileName());
+                jsonScanReportFile.addProperty("filePath", filePath);
+                String fileContent = "";
+                try {
+                    fileContent = Files.readString(Path.of(filePath));
+                } catch (IOException ex) {
+                    throw new RuntimeException(ex);
+                }
+                jsonScanReportFile.addProperty("fileContent", fileContent);
+
+                JsonObject jsonScanReportIssueTextRange = new JsonObject();
+                LineRange lineRange = issueIml.location().lineRange();
+                jsonScanReportIssueTextRange.addProperty("startLine", lineRange.startLine().line());
+                jsonScanReportIssueTextRange.addProperty("startLineOffset", lineRange.startLine().offset());
+                jsonScanReportIssueTextRange.addProperty("endLine", lineRange.endLine().line());
+                jsonScanReportIssueTextRange.addProperty("endLineOffset", lineRange.endLine().offset());
```

Comment:
Shall we extract this out to a function because this is done in both `if` and `else` clauses.